tests/functional/io_apis_test.c

```diff
@@ -0,0 +1,831 @@
+// Copyright 2018-2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * ----------------------------------------------------------------------------
+ * splinter_io_apis_test.c --
```

Comment:
New test case to exercise core IO APIs.

For now, this is not adding any great testing value, other than added value to my understanding of the IO sub-system.

If ever we bring-on a new platform [ E.g. Async-IO on Mac/OSX] this test will provide regression cover.